packages/squiggle-lang/src/rescript/FunctionRegistry/Library/FR_Danger.res

```diff
@@ -0,0 +1,178 @@
+      (),
+    )
+}
+
+module FunctionToNumberZero = {
+  let make = (name, fn) =>
+    FnDefinition.make(
+      ~name,
+      ~inputs=[FRTypeLambda],
+      ~run=(_, inputs, _, _) => {
+        Ok(0.0)->E.R2.fmap(Wrappers.evNumber)
+      },
+      (),
+    )
+}
+
+module Internals = {
+  let factorial = Stdlib.Math.factorial
+  let choose = ((n, k)) => factorial(n) /. (factorial(n -. k) *. factorial(k))
+  let pow = (base, exp) => Js.Math.pow_float(~base, ~exp)
+  let binomial = ((n, k, p)) => choose((n, k)) *. pow(p, k) *. pow(1.0 -. p, n -. k)
+  let applyFunctionAtPoint = (
+    aLambda,
+    internalNumber: internalExpressionValue,
+    environment,
+    reducer,
+  ): result<ReducerInterface_InternalExpressionValue.t, Reducer_ErrorValue.errorValue> => {
+    let x = internalNumber
+    let result = Reducer_Expression_Lambda.doLambdaCall(aLambda, list{x}, environment, reducer)
+    result
+  }
+  let internalZero = ReducerInterface_InternalExpressionValue.IEvNumber(0.0)
+  let applyFunctionAtZero = (aLambda, environment, reducer) =>
+    applyFunctionAtPoint(aLambda, internalZero, environment, reducer)
+  let applyFunctionAtFloat = (aLambda, point, environment, reducer) =>
+    applyFunctionAtPoint(aLambda, ReducerInterface_InternalExpressionValue.IEvNumber(point))
+  let integrateFunction = (aLambda, min: float, max: float, increment, environment, reducer) => {
+    // Should be easy, but tired today.
+    0.0
+  }
+  let getDiminishingMarginalReturnsEquilibrium = "To do"
+}
+
+let library = [
+  Function.make(
+    ~name="laplace",
+    ~nameSpace,
+    ~requiresNamespace,
+    ~output=EvtNumber,
+    ~examples=[`Danger.laplace(1, 20)`],
+    ~definitions=[
+      TwoNumbersToNumber.make("laplace", ((successes, trials)) =>
+        (successes +. 1.0) /. (trials +. 2.0)
+      ),
+    ],
+    (),
+  ),
+  Function.make(
+    ~name="factorial",
+    ~nameSpace,
+    ~requiresNamespace,
+    ~output=EvtNumber,
+    ~examples=[`Danger.factorial(20)`],
+    ~definitions=[NumberToNumber.make("factorial", Internals.factorial)],
+    (),
+  ),
+  Function.make(
+    ~name="choose",
+    ~nameSpace,
+    ~requiresNamespace,
+    ~output=EvtNumber,
+    ~examples=[`Danger.choose(1, 20)`],
+    ~definitions=[TwoNumbersToNumber.make("choose", Internals.choose)],
+    (),
+  ),
+  Function.make(
+    ~name="binomial",
+    ~nameSpace,
+    ~requiresNamespace,
+    ~output=EvtNumber,
+    ~examples=[`Danger.binomial(1, 20, 0.5)`],
+    ~definitions=[ThreeNumbersToNumber.make("binomial", Internals.binomial)],
+    (),
+  ),
+  Function.make(
+    ~name="functionToZero",
+    ~nameSpace,
+    ~requiresNamespace,
+    ~output=EvtNumber,
+    ~examples=[`Danger.functionToZero({|x| x})`],
+    ~definitions=[FunctionToNumberZero.make("functionToZero", x => x)],
+    (),
+  ),
+  Function.make(
+    ~name="applyFunctionAtZero",
+    ~nameSpace,
+    ~output=EvtArray,
+    ~requiresNamespace=false,
+    ~examples=[`Danger.applyFunctionAtZero({|x| x+1})`],
```

Comment:
What's the use of this and ``applyFunctionAtPoint``?

Reply:
I was building up to an integral function, and this was useful to figure it out on the way.